ingestion/src/metadata/ingestion/source/pipeline/gluepipeline/metadata.py

```diff
@@ -145,9 +180,95 @@ def get_downstream_tasks(self, task_unique_id, tasks):
+                            table_model = TABLE_MODEL_MAP[key].model_validate(entity)
+                        elif "Catalog" in key:
+                            table_model = CatalogSource.model_validate(entity)
+                        elif key in STORAGE_MODEL_MAP:
+                            storage_model = STORAGE_MODEL_MAP[key].model_validate(
+                                entity
+                            )
+                        if table_model:
+                            for db_service_name in self.get_db_service_names():
+                                table_entity = self.metadata.get_by_name(
+                                    entity=Table,
+                                    fqn=fqn.build(
+                                        metadata=self.metadata,
+                                        entity_type=Table,
+                                        table_name=table_model.table_name,
+                                        database_name=table_model.database_name,
+                                        schema_name=table_model.schema_name,
+                                        service_name=db_service_name,
+                                    ),
+                                )
+                                if table_entity:
+                                    if key.endswith("Source"):
+                                        lineage_details["sources"].append(
+                                            (table_entity, "table")
+                                        )
+                                    else:
+                                        lineage_details["targets"].append(
+                                            (table_entity, "table")
+                                        )
+                                    break
+                        if storage_model:
+                            for (
+                                storage_service_name
+                            ) in self.get_storage_service_names():
+                                for path in storage_model.Paths or [storage_model.Path]:
+                                    storage_entity = self.metadata.get_by_name(
+                                        entity=Table,
+                                        fqn=fqn.build(
+                                            self.metadata,
+                                            entity_type=Container,
+                                            service_name=storage_service_name,
+                                            parent_container="",
+                                            container_name=path.split("/")[-1]
+                                            if "/" in path
+                                            else None,
+                                        ),
+                                    )
+                                    if storage_entity:
+                                        if key.endswith("Source"):
```

Comment:
again here we are building fqn for container but trying to get table by name, how would this work??